### PR TITLE
Fix access to default backend for 404 pages

### PR DESCRIPTION
**Pull request by sjmiller609**
## Description

Fix a test that is failing in CI and also fix network access from nginx ingress to default backend

## 🎟 Issue(s)

no issue associated, part of release 0.20 fixes

## 🧪  Testing

this change only adjusts tests or includes a functional test for the network policy change.

## 📸 Screenshots

> Add screenshots to illustrate the changes, if applicable.

## 📋 Checklist

- [x] The PR title is informative to the user experience

### CODE REVIEW

File: bin/functional-tests/conftest.py [hunk 1]
```diff
@@ -21,7 +21,27 @@ def create_kube_client(in_cluster=False):
...
+    namespace = os.environ.get('NAMESPACE')
+    release_name = os.environ.get('RELEASE_NAME')
+    if not namespace:
+        print("NAMESPACE env var is not present, using 'astronomer' namespace")
+        namespace = 'astronomer'
+    if not release_name:
+        print("RELEASE_NAME env var is not present, assuming 'astronomer' is the release name")
+        release_name = 'astronomer'
+    kube = create_kube_client()
+    pods = kube.list_namespaced_pod(namespace, label_selector=f"component=ingress-controller")
```
Comment (danielhoherd) [inline bin/functional-tests/conftest.py:39]:
```suggestion
    pods = kube.list_namespaced_pod(namespace, label_selector="component=ingress-controller")
```

No need for f-string.

File: bin/functional-tests/test_config.py [hunk 2]
```diff
@@ -33,7 +33,10 @@ def test_houston_config(houston_api):
             f"Expected not to find 'localhost' in the 'servers' configuration. Found:\n\n{houston_config['nats']}"
 
 def test_houston_can_reach_prometheus(houston_api):
-    houston_api.check_output("wget --timeout=1 http://astronomer-prometheus.astronomer.svc.cluster.local:9090/targets")
+    houston_api.check_output("wget -qO- --timeout=1 http://astronomer-prometheus.astronomer.svc.cluster.local:9090/targets")
+
+def test_nginx_can_reach_default_backend(nginx):
+    nginx.check_output("curl --max-time 1 http://astronomer-nginx-default-backend:8080")
```
Comment (jpweber) [inline bin/functional-tests/test_config.py:39]: super tiny nit. Do you want to add `-s` here for silent?

**Review by jpweber — APPROVED** [review]

**Review by danielhoherd — APPROVED** [review]
Just one small change, otherwise LGTM.

File: bin/functional-tests/test_config.py (hunk 2 above)
Comment (danielhoherd) [reply bin/functional-tests/test_config.py:39]: Also, why wget in one instance and wget in another? It seems like we should choose one method for doing these checks and stick with it.

Comment (sjmiller609) [reply bin/functional-tests/test_config.py]: because of what is available in the container

**Pull request merged**